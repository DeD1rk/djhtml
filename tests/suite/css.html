<!-- Basic CSS -->
<style>
    a {
        text-decoration: none;
    }
</style>


<!-- Nested SCSS -->
<style>
    a {
        text-decoration: none;
        &:hover {
            text-decoration: underline;
        }
    }
</style>


<!-- Multiline comment -->
<style>
    /*
     * Hi mom!
     *
     */
</style>


<!-- CSS Function -->
<style>
    #gradient {
        background: linear-gradient(
                        to top left,
                        rgba(204,0,0,1),
                        rgba(204,0,0,0)
                    );
    }
</style>


<!-- @font-face typical usage -->
<style>
    @font-face {
        font-family: 'Open Sans';
        font-style: normal;
        font-weight: normal;
        src: url('Open_Sans.woff2') format('woff2'),
             url('Open_Sans.woff') format('woff'),
             url('Open_Sans.eot?#iefix') format('embedded-opentype'),
             url('Open_Sans.svg#OpenSans') format('svg'),
             url('Open_Sans.ttf') format('truetype');
    }
</style>


<!-- @media -->
<style>
    @media(max-width: 800px) {
        body {
            margin-top: 1em;
            margin-bottom: 1em;
        }
    }
</style>


<!-- Missing semicolons -->
<style>
    p { align: left }
    h1 {
        font-weight: bold;
        font-size: huge
    }
    footnote {
        dontread: true !important
    }
</style>


<!-- Nested SCSS with missing semicolons and different indentation styles -->
<style>
    article {
        p { color: #999; align: left }
    }
    article {
        p { color: #999;
            align: left }
    }
    article {
        p {
            color: #999;
            align: left }
    }
    article {
        p { color: #999;
            align: left
        }
    }
    article {
        p {
            color: #999;
            align: left
        }
    }
    article {
        p
        {
            color: #999;
            align: left;
        }
    }
</style>


<!-- Selector weirdness -->
<style>
    blockquote:before {
        content: "text-align: right";
    }
    @media(text-align: right) {
        a:hover, a:active {
            text-decoration:none
        }
        p::first-line {
            color: red;
        }
    }
</style>


<!-- Nested SCSS with multi-line statements -->
<style>
    a:hover, a:active {
        foo: bar,
             baz;

        a:hover, a:active {
            foo: bar
                 baz
        }
    }
</style>


<!-- The difference between multi-line statements and nested selectors -->
<style>
    body {
        font-family: Arial,
                     Helvetica;
        a:hover,
        a:active {
            color: red;
        }
    }
</style>
