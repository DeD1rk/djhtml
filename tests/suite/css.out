<!-- Basic CSS -->
<style>
    a {
        text-decoration: none;
    }
</style>


<!-- Nested SCSS -->
<style>
    a {
        text-decoration: none;
        &:hover {
            text-decoration: underline;
        }
    }
</style>


<!-- Multiline comment -->
<style>
    /*
     * Hi mom!
     *
     */
</style>


<!-- CSS Function -->
<style>
    #gradient {
        background: linear-gradient(
            to top left,
            rgba(204,0,0,1),
            rgba(204,0,0,0)
        );
    }
</style>
